### PR TITLE
Add configuration docs

### PR DESCRIPTION
**Pull request by ChrisPenner**
We didn't have a place to document all of the ucm env vars and configuration, now we do 😄

### CODE REVIEW

**Comment by ChrisPenner** [comment]
cc @rlmark  just so you know this exists 😄

**Review by pchiusano — APPROVED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+E.g.
+
+```sh
+# Enable ALL debugging flags (likely quite noisy)
+$ UNISON_DEBUG= ucm
+# Enable timing debugging, printing how long different actions take.
+$ UNISON_DEBUG=TIMING ucm
+# Enable LSP and TIMING debugging
+$ UNISON_DEBUG=LSP,TIMING ucm
```
Comment (aryairani) [inline docs/configuration.md]:
```suggestion
# Enable ALL debugging flags (likely quite noisy)
$ UNISON_DEBUG= ucm
# Enable timing debugging, printing how long different actions take.
$ UNISON_DEBUG=timing ucm
# Enable LSP and TIMING debugging
$ UNISON_DEBUG=lsp,timing ucm
```

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+### `UNISON_DEBUG`
+
+Enable debugging output for various portions of the application. 
+See `lib/unison-prelude/src/Unison/Debug.hs` for the full list of supported flags.
+
+E.g.
+
+```sh
+# Enable ALL debugging flags (likely quite noisy)
+$ UNISON_DEBUG= ucm
```
Comment (aryairani) [inline docs/configuration.md:24]: Side note, this probably should disable all debugging, not enable all. I'd suggest `UNISON_DEBUG=all` for all.

**Review by aryairani — APPROVED** [review]

File: docs/configuration.md (hunk 2 above)
Comment (ChrisPenner) [reply docs/configuration.md:24]: I have a preference either way, IIRC I had `all` in the original design and @mitchellwrosen requested this behaviour so maybe talk it over between the two of you.

File: docs/configuration.md (hunk 1 above)
Comment (ChrisPenner) [reply docs/configuration.md]: Any particular reason for the change? I don't really care either way, but I suspect you suggested it for a reason?

**Pull request merged**